[CI] MlWithSecurityIT test {yaml=ml/data_frame_analytics_crud/Test get stats given multiple analytics} failing

**Build Scans:**
- [elasticsearch-intake #40397 / part4](https://gradle-enterprise.elastic.co/s/wgbx6tyxpejug)

**Reproduction Line:**
```
./gradlew ":x-pack:plugin:ml:qa:ml-with-security:yamlRestTest" --tests "org.elasticsearch.smoketest.MlWithSecurityIT.test {yaml=ml/data_frame_analytics_crud/Test get stats given multiple analytics}" -Dtests.seed=F7A0DF097BDBE18C -Dtests.locale=luy-KE -Dtests.timezone=Asia/Rangoon -Druntime.java=26
```

**Applicable branches:**
9.3

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.smoketest.MlWithSecurityIT),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!('test%20%7Byaml%3Dml%2Fdata_frame_analytics_crud%2FTest%20get%20stats%20given%20multiple%20analytics%7D'),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: org.junit.internal.runners.model.MultipleFailureException: There were 5 errors:
  java.lang.AssertionError(expected [2xx] status code but api [ml.get_data_frame_analytics_stats] returned [500 Internal Server Error] [{"error":{"root_cause":[{"type":"no_shard_available_action_exception","reason":"[yamlRestTest-0][127.0.0.1:41531][indices:data/read/search[phase/query]]","stack_trace":"org.elasticsearch.action.NoShardAvailableActionException: [yamlRestTest-0][127.0.0.1:41531][indices:data/read/search[phase/query]]\n"}],"type":"exception","reason":"all shards failed","caused_by":{"type":"search_phase_execution_exception","reason":"all shards failed","phase":"query","grouped":true,"failed_shards":[{"shard":0,"index":".ml-stats-000001","node":"llQEP3LSRzCOmIoMHDvJ7g","reason":{"type":"no_shard_available_action_exception","reason":"[yamlRestTest-0][127.0.0.1:41531][indices:data/read/search[phase/query]]","stack_trace":"org.elasticsearch.action.NoShardAvailableActionEx
[truncated]
```

**Issue Reasons:**
- [9.3] 1 consecutive failures in test test {yaml=ml/data_frame_analytics_crud/Test get stats given multiple analytics}
- [9.3] 1 consecutive failures in step part4
- [9.3] 1 consecutive failures in pipeline elasticsearch-intake

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -327,9 +327,6 @@ tests:
 - class: org.elasticsearch.smoketest.SmokeTestWatcherTestSuiteIT
   method: testMonitorClusterHealth
   issue: https://github.com/elastic/elasticsearch/issues/148615
-- class: org.elasticsearch.smoketest.MlWithSecurityIT
-  method: test {yaml=ml/data_frame_analytics_crud/Test get stats given multiple analytics}
-  issue: https://github.com/elastic/elasticsearch/issues/150078
 - class: org.elasticsearch.xpack.rank.rrf.RRFRankClientYamlTestSuiteIT
   method: test {yaml=rrf/700_rrf_retriever_search_api_compatibility/nested compound retrievers with min_score correctly compute total_hits}
   issue: https://github.com/elastic/elasticsearch/issues/150086

--- a/x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/action/TransportGetDataFrameAnalyticsStatsAction.java
+++ b/x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/action/TransportGetDataFrameAnalyticsStatsAction.java
@@ -10,10 +10,13 @@ import org.apache.logging.log4j.LogManager;
 import org.apache.logging.log4j.Logger;
 import org.elasticsearch.action.ActionListener;
 import org.elasticsearch.action.FailedNodeException;
+import org.elasticsearch.action.NoShardAvailableActionException;
 import org.elasticsearch.action.TaskOperationFailure;
 import org.elasticsearch.action.search.MultiSearchRequest;
 import org.elasticsearch.action.search.MultiSearchResponse;
+import org.elasticsearch.action.search.SearchPhaseExecutionException;
 import org.elasticsearch.action.search.SearchRequest;
+import org.elasticsearch.action.search.ShardSearchFailure;
 import org.elasticsearch.action.search.TransportMultiSearchAction;
 import org.elasticsearch.action.support.ActionFilters;
 import org.elasticsearch.action.support.IndicesOptions;
@@ -52,6 +55,7 @@ import org.elasticsearch.xpack.core.ml.dataframe.stats.common.MemoryUsage;
 import org.elasticsearch.xpack.core.ml.dataframe.stats.outlierdetection.OutlierDetectionStats;
 import org.elasticsearch.xpack.core.ml.dataframe.stats.regression.RegressionStats;
 import org.elasticsearch.xpack.core.ml.job.persistence.AnomalyDetectorsIndex;
+import org.elasticsearch.xpack.core.ml.job.persistence.AnomalyDetectorsIndexFields;
 import org.elasticsearch.xpack.core.ml.utils.ExceptionsHelper;
 import org.elasticsearch.xpack.core.ml.utils.PhaseProgress;
 import org.elasticsearch.xpack.ml.dataframe.DataFrameAnalyticsTask;
@@ -275,6 +279,18 @@ public class TransportGetDataFrameAnalyticsStatsAction extends TransportTasksAct
                     MultiSearchResponse.Item itemResponse = itemResponses[i];
                     if (itemResponse.isFailure()) {
                         SearchRequest itemRequest = multiSearchRequest.requests().get(i);
+                        if (recoverableMlShardFailureForStatsSearch(itemRequest, itemResponse.getFailure())) {
+                            logger.warn(
+                                () -> format(
+                                    "[{}] Skipping unreadable ML stats search while shards are unavailable; "
+                                        + "assuming no persisted stats for this sub-request [indices={}]: {}",
+                                    config.getId(),
+                                    Arrays.toString(itemRequest.indices()),
+                                    itemResponse.getFailureMessage()
+                                )
+                            );
+                            continue;
+                        }
                         logger.error(
                             () -> format(
                                 "[%s] Item failure encountered during multi search for request [indices=%s, source=%s]: %s",
@@ -401,5 +417,49 @@ public class TransportGetDataFrameAnalyticsStatsAction extends TransportTasksAct
         private RetrievedStatsHolder(List<PhaseProgress> defaultProgress) {
             progress = new StoredProgress(defaultProgress);
         }
+    }
+
+    /**
+     * When {@link TransportStartDataFrameAnalyticsAction} starts a job, it loads stored progress from
+     * {@link MlStatsIndex} and {@link AnomalyDetectorsIndex} before validating the source data. If those ML
+     * system indices exist but shards are not yet assigned (e.g. shortly after index creation), multi-search
+     * fails with {@link NoShardAvailableActionException}. Treat that like missing stats documents so
+     * validation can proceed with default progress (see #138409).
+     */
+    private static boolean recoverableMlShardFailureForStatsSearch(SearchRequest request, Exception failure) {
+        if (targetsMlStatsOrStateIndices(request) == false) {
+            return false;
+        }
+        return isShardNotAvailableFailure(failure);
+    }
+
+    private static boolean targetsMlStatsOrStateIndices(SearchRequest request) {
+        for (String index : request.indices()) {
+            if (index.contains(MlStatsIndex.TEMPLATE_NAME) || index.contains(AnomalyDetectorsIndexFields.STATE_INDEX_PREFIX)) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    private static boolean isShardNotAvailableFailure(Throwable failure) {
+        Throwable current = failure;
+        while (current != null) {
+            if (current instanceof NoShardAvailableActionException) {
+                return true;
+            }
+            if (current instanceof SearchPhaseExecutionException spe) {
+                ShardSearchFailure[] shardFailures = spe.shardFailures();
+                if (shardFailures != null) {
+                    for (ShardSearchFailure shardFailure : shardFailures) {
+                        if (ExceptionsHelper.unwrapCause(shardFailure.getCause()) instanceof NoShardAvailableActionException) {
+                            return true;
+                        }
+                    }
+                }
+            }
+            current = current.getCause();
+        }
+        return false;
     }
 }

